protocol/adapter/datasync_whisper.go

```diff
@@ -95,8 +102,64 @@ func (w *DataSyncWhisperAdapter) decodePayload(message *whisper.ReceivedMessage)
 	return
 }
 
-func (w *DataSyncWhisperAdapter) decodeMessages(payload protobuf.Payload) []*protocol.Message {
-	messages := make([]*protocol.Message, 0)
+func (w *DataSyncWhisperAdapter) OnNewMessages(messages []*msgfilter.Messages) {
+	for _, filterMessages := range messages {
+
+		receivedMessages := &protocol.ReceivedMessages{}
+
+		var options protocol.ChatOptions
+
+		if filterMessages.Chat.OneToOne || filterMessages.Chat.Negotiated || filterMessages.Chat.Discovery {
```

Comment:
I can't find a place where `OntToOne` is set to `true` ever.

Reply:
It might be passed by the client, probably we should make that more explicit